google-cloud-speech/lib/google/cloud/speech/credentials.rb

```diff
@@ -12,19 +12,12 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-require "googleauth"
+require "google/cloud/speech/v1/credentials"
```

Comment:
This gem isn't 1.0 yet, and the next release is definitely a breaking change. Do we really need to maintain this file and constant? For GA gems yes, but should we do that here?

@quartzmo what do you think?

Reply:
The constant being `Google::Cloud::Speech::Credentials`.

Reply:
If `google/cloud/speech/credentials.rb` can be removed from the public API, this would be a good time to do it, wouldn't it?

Reply:
I think it can be, yes.

I just checked all the code examples and I don't see any examples of how to use the environment variables or the Credentials classes. But perhaps that should be solved in a separate change?

Reply:
Hmmm, yes, that would probably be needed in other gems as well, correct?

Reply:
Yeah, probably a good idea. The docs site has the general authentication guide, but the GAPIC generated gems should probably list what environment variables are used somewhere in the gem documentation.